src/hooks/useIsSnapInstalled/useIsSnapInstalled.tsx

```diff
@@ -156,3 +159,37 @@ export const useIsSnapInstalled = (): null | boolean => {
 
   return isSnapInstalled
 }
+
+export const canAddMetaMaskAccount = ({
+  accountNumber,
+  chainId,
+  wallet,
+  isSnapInstalled,
+}: {
+  accountNumber: number
+  chainId: ChainId
+  wallet: HDWallet
+  isSnapInstalled: boolean
+}) => {
+  // Can always add 0th account regardless of chain/snap installation
+  if (accountNumber === 0) return true
+
+  const isMetaMaskMultichainWallet = wallet instanceof MetaMaskShapeShiftMultiChainHDWallet
```

Comment:
wallet check doesn't really serve a purpose in the context of `canAddMetaMaskAccount` since the implication is metamask already.

Reply:
Yeah consumers should always check for the correct adapter beforehand so might as well remove it, or perhaps even better, throw since this should never happen and if it does, it means we did something wrong

Reply:
6fd3f6f81c